File: Backend/BaseMgtService/Senior.AgileAI.BaseMgt.Application/DTOs/ProjectMemberDTO.cs
namespace Senior.AgileAI.BaseMgt.Application.DTOs
{
    public class ProjectMemberDTO
    {
        public Guid UserId { get; set; }
        public Guid MemberId { get; set; }
        public required string Name { get; set; }
        public required string Email { get; set; }
        public required bool IsAdmin { get; set; }
        public required string Meetings { get; set; }
        public required string Members { get; set; }
        public required string Requirements { get; set; }
        public required string Tasks { get; set; }
        public required string Settings { get; set; }
    }
}

File: agilemeets_mobile/lib/data/models/project/project_member_dto.dart
class ProjectMemberDTO {
  final String userId;
  final String memberId;
  final String name;
  final String email;
  final bool isAdmin;
  final String meetings;
  final String members;
  final String requirements;
  final String tasks;
  final String settings;

  ProjectMemberDTO({
    required this.userId,
    required this.memberId,
    required this.name,
    required this.email,
    required this.isAdmin,
    required this.meetings,
    required this.members,
    required this.requirements,
    required this.tasks,
    required this.settings,
  });

  factory ProjectMemberDTO.fromJson(Map<String, dynamic> json) {
    return ProjectMemberDTO(
      userId: json['userId'] ?? '',
      memberId: json['memberId'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      isAdmin: json['isAdmin'] ?? false,
      meetings: json['meetings'] ?? 'None',
      members: json['members'] ?? 'None',
      requirements: json['requirements'] ?? 'None',
      tasks: json['tasks'] ?? 'None',
      settings: json['settings'] ?? 'None',
    );
  }

  Map<String, dynamic> toJson() => {
    'userId': userId,
    'memberId': memberId,
    'name': name,
    'email': email,
    'isAdmin': isAdmin,
    'meetings': meetings,
    'members': members,
    'requirements': requirements,
    'tasks': tasks,
    'settings': settings,
  };
}